### PR TITLE
feat: add support for Tracker resource pagination

### PR DESCRIPTION
**Pull request by JohanGHole**
* Modified the `hasNext` and `fetchPage` methods in `PageIterable` to determine the existence of additional pages and fetch them accordingly. It addresses the issue of handling cases where the total number of items is a multiple of pageSize, ensuring correct iteration through all pages of tracker data.
* Added tests for single page and multiple page collections for both Tracker and Aggregate endpoints. 
* Added AFI metadata to test Tracker endpoint.
* Updated test container from version 2.37 to 2.38.

### CODE REVIEW

File: sdk/src/main/java/org/hisp/dhis/api/model/Page.java [hunk 1]
```diff
@@ -65,4 +107,20 @@ public void setAdditionalProperty( String name, Object value )
...
+            if ( page == null )
+            {
+                page = pager.getPage();
+            }
+            if ( pageSize == null )
+            {
+                pageSize = pager.getPageSize();
+            }
+        }
+    }
```
Comment (claudemamo) [inline sdk/src/main/java/org/hisp/dhis/api/model/Page.java:124]: What is this code for? I set a breakpoint here but the tests never stepped into this method.

File: sdk/src/test/java/org/hisp/dhis/integration/sdk/internal/operation/page/PagingCollectOperationTestCase.java [hunk 2]
```diff
@@ -42,8 +44,9 @@
 public class PagingCollectOperationTestCase extends AbstractTestCase
 {
     @Test
-    public void testTransfer()
+    public void testOnePageAggregateTransfer()
```
Comment (claudemamo) [inline sdk/src/test/java/org/hisp/dhis/integration/sdk/internal/operation/page/PagingCollectOperationTestCase.java:47]: The name is misleading here. Org units are not "aggregate" per se. Why not just `testOnePageTransfer`? As a side, note, I prefer naming tests after the method under test and then append the scenario description to the test name. For example: `testTransferGivenSinglePage`. But to each his own.

File: sdk/src/main/java/org/hisp/dhis/api/model/Page.java (hunk 1 above)
Comment (JohanGHole) [reply sdk/src/main/java/org/hisp/dhis/api/model/Page.java:124]: Yes you are correct. This became redundant after introducing the `lastPage` flag to the `PageIterable`. 
Removed.

File: sdk/src/test/java/org/hisp/dhis/integration/sdk/internal/operation/page/PagingCollectOperationTestCase.java (hunk 2 above)
Comment (JohanGHole) [reply sdk/src/test/java/org/hisp/dhis/integration/sdk/internal/operation/page/PagingCollectOperationTestCase.java:47]: Ok fair enough. I will change the names to reflect the convention.

**Pull request merged**